MediaBrowser.Model/Providers/ExternalIdInfo.cs

```diff
@@ -16,6 +16,12 @@ public class ExternalIdInfo
         /// <value>The key.</value>
         public string Key { get; set; }
 
+        /// <summary>
+        /// Gets or sets the type.
+        /// </summary>
+        /// <value>The type.</value>
+        public string Type { get; set; }
```

Comment:
I think the `Key` property is already acts as a unique identifier for the `ExternalIdInfo` (based on [this](https://github.com/jellyfin/jellyfin-web/issues/900#issuecomment-596123103) issue tread/comment) so why add this property? Why can't the client just do the translation based on the `Key`?

Reply:
The key should indeed be unique.

Reply:
It seems we need two pieces of information to make a unique key:

1. The provider name (IE: TMDB, MusicBrainz, etc)
2. The media type (Movie, Artist, etc)

The provider name does not need to be localized but the media type does. We can certainly send these two pieces of information to the client in one field but they would have to be pulled apart on the client or we would have to have localized strings for each provider/type pair.

So if the key needs to be unique, we can build a unique key on the server (using the provider/type pair). This would never need to be localized or displayed in the UI.

But in order to build the Label in the UI, we still need the provider name (not localized) and the media type (localized on the client)

Does this seem correct? Can we translate correctly on the client using just the id field?

Reply:
@lfoust Okay, yes that does make sense to me 👍

Reply:
@mark-monteiro Great! I'll push an update to make sure the ids are unique on the server side

Reply:
@Bond-009 you have had good ideas about the external IDs before, what are your thoughts?

Reply:
@dkanada I looked into the uniqueness of ids and I noticed that they currently are not unique. For example, all of TheTVDB externalIds have the same `Id`. I can make them unique but I am not sure what this can break. I could also make a 'further clean up ExternalIds' work item and address this stuff (including adding tests?) in a separate checkin? What do you think?